internal/app/service/create_reservation.go

```diff
@@ -0,0 +1,120 @@
+package service
+
+import (
+	"context"
+	"time"
+
+	"github.com/sirupsen/logrus"
+
+	"github.com/FurmanovD/roommgr/internal/pkg/db/automodel"
+	api "github.com/FurmanovD/roommgr/pkg/api/v1"
+)
+
+func (s *serviceImpl) CreateReservation(
```

Comment:
the function is too long, consider refactoring

Reply:
done